cron/src/tests/test_message_service.py

```diff
@@ -1,49 +1,43 @@
 import logging
 from datetime import datetime
-from pytest import fixture
+import pytest
 from freezegun import freeze_time
 from notifications.message_service import MessageService
 from notifications.definitions import RecordInfo, BookInfo, BorrowerInfo
 
 
 class TestMessageService():
-    def test_compose_messages_senders(self, test_data):
+    @pytest.mark.parametrize(
+        'test_method, lenght',
+        [('compose_user_messages', 2), ('compose_admin_messages', 1)]
+    )
+    def test_compose_messages_senders(self, test_data, test_method, lenght):
         sender = 'foo <foo@example.com>'
         template = '{{recipient_surname}}, {{recipient_name}}'
         service = MessageService(sender)
-        messages = list(service.compose_messages(template, test_data))
-
-        def recipient_match(message, recipient_name, recipient_surname):
-            return str(message.get_content()) == '{}, {}\n'.format(
-                recipient_surname,
-                recipient_name)
+        messages = list(getattr(service, test_method)(template, test_data))
```

Comment:
Why do parametrize test with function name you want to call and then use reflection to call required function? I would rather have two separate tests with no redundant magic.